gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/iterator/IteratorUtils.java

```diff
@@ -116,39 +120,64 @@ public S next() {
                     throw FastNoSuchElementException.instance();
                 return iterator.next();
             }
+
+            @Override
+            public void close() {
+                CloseableIterator.closeIterator(iterator);
+            }
         };
     }
 
     ///////////////////
 
     public static <T> boolean allMatch(final Iterator<T> iterator, final Predicate<T> predicate) {
-        while (iterator.hasNext()) {
-            if (!predicate.test(iterator.next())) {
-                return false;
+        try {
```

Comment:
could you explain this extra safety a bit please? were you running into some scenario where the iterator was throwing an exception?

Reply:
It's not about exceptions - this class was wrapping iterators inside iterators but not taking proper care / responsibility for closing them. These changes just make sure that one iterator wraps another iterator it closes it properly.